File: Lab2/go/task2.go
package main

func countVowelsConsonants(data []byte, count []int) {
    for _, c := range string(data) {
        switch c {
        case 'a', 'A', 'á', 'Á':
            count[1]++
        case 'e', 'E', 'é', 'É':
            count[2]++
        case 'i', 'I', 'í', 'Í':
            count[3]++
        case 'o', 'O', 'ó', 'Ó':
            count[4]++
        case 'u', 'U', 'ú', 'Ú':
            count[5]++
        case ' ':
            continue 
        default:
            count[6]++
        }
    }

    count[0] = 0 // Aseguramos que empiece en cero
    for i := 1; i <= 5; i++ {
        count[0] += count[i]
    }
}

File: Lab2/c/task2.c
void count_vowels_consonants(const char *input, int *count) {
    for (const char *p = input; *p; p++) {
        switch (*p) {
            case 'a': case 'A': count[1]++; break;
            case 'e': case 'E': count[2]++; break;
            case 'i': case 'I': count[3]++; break;
            case 'o': case 'O': count[4]++; break;
            case 'u': case 'U': count[5]++; break;
            case ' ': break; 
            default: count[6]++; break;
        }
    }

    for (int i = 1; i <= 5; i++) {
        count[0] += count[i];
    }
}